shared-libs/settings/src/index.js

```diff
@@ -1,39 +1,96 @@
 const request = require('request-promise-native');
-const RESULT_PARSE_REGEX = /^"(.*)"\n?$/;
+const crypto = require('crypto');
 
-// This API gives weird psuedo-JSON results:
-//   "password"\n
-// Should be just `password`
-const parseResponse = response => response.match(RESULT_PARSE_REGEX)[1];
+const IV_LENGTH = 16;
+const CRYPTO_ALGO = 'aes-256-cbc';
 
-const getCouchNodeName = () => process.env.COUCH_NODE_NAME;
+const getCredentialId = id => `credential:${id}`;
 
-const getCredentials = (key) => {
-  try {
-    const couchConfigUrl = getCouchConfigUrl();
-    return request
-      .get(`${couchConfigUrl}/medic-credentials/${key}`)
-      .then(parseResponse)
-      .catch(err => {
-        if (err.statusCode === 404) {
-          // No credentials defined
-          return;
-        }
-        // Throw it regardless so the process gets halted, we just error above for higher specificity
-        throw err;
-      });
+const getCouchNodeName = () => process.env.COUCH_NODE_NAME;
 
-  } catch (error) {
-    return Promise.reject(error);
-  }
+const getCouchUrl = () => {
+  const couchUrl = process.env.COUCH_URL;
+  return couchUrl && couchUrl.replace(/\/$/, '');
 };
 
 const getServerUrl = () => {
-  if (!process.env.COUCH_URL) {
-    return;
+  const couchUrl = process.env.COUCH_URL;
+  return couchUrl && couchUrl.slice(0, couchUrl.lastIndexOf('/'));
+};
+
+const getVaultUrl = (id) => `${getCouchUrl()}-vault/${getCredentialId(id)}`;
+
+const getCredentialsDoc = (id) => {
+  if (!id) {
+    return Promise.reject(new Error('You must pass the key for the credentials you want'));
   }
-  const couchUrl = process.env.COUCH_URL.replace(/\/$/, '');
-  return couchUrl.slice(0, couchUrl.lastIndexOf('/'));
+  return request
+    .get(getVaultUrl(id), { json: true })
+    .catch(err => {
+      if (err.statusCode === 404) {
+        // No credentials defined
+        return;
+      }
+      // Throw it regardless so the process gets halted, we just error above for higher specificity
+      throw err;
+    });
+};
+
+const getKey = () => {
+  // NB: This path will need to change when we upgrade to CouchDB v3.2
+  // https://docs.couchdb.org/en/stable/config/auth.html#chttpd_auth/secret
+  const url = `${getCouchConfigUrl()}/couch_httpd_auth/secret`;
+  return request.get(url, { json: true });
+};
+
+const encrypt = (text) => {
+  return getKey()
+    .then(key => {
+      const iv = crypto.randomBytes(IV_LENGTH);
+      const cipher = crypto.createCipheriv(CRYPTO_ALGO, Buffer.from(key), iv);
```

Comment:
I'm trying this locally, with Node 16, and I'm getting this error: 
```
2022-06-10 08:05:09 [33mWARN[39m: Non-numeric error code: ERR_CRYPTO_INVALID_KEYLEN 
2022-06-10 08:05:09 [31mERROR[39m: Server error: RangeError: Invalid key length
    at Cipheriv.createCipherBase (node:internal/crypto/cipher:116:19)
    at Cipheriv.createCipherWithIV (node:internal/crypto/cipher:135:3)
    at new Cipheriv (node:internal/crypto/cipher:243:3)
    at Object.createCipheriv (node:crypto:138:10)
    at /shared-libs/settings/src/index.js:49:29
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Promise.all (index 1) {
  [stack]: 'RangeError: Invalid key length\n' +
    '    at Cipheriv.createCipherBase (node:internal/crypto/cipher:116:19)\n' +
    '    at Cipheriv.createCipherWithIV (node:internal/crypto/cipher:135:3)\n' +
    '    at new Cipheriv (node:internal/crypto/cipher:243:3)\n' +
    '    at Object.createCipheriv (node:crypto:138:10)\n' +
    '    at /shared-libs/settings/src/index.js:49:29\n' +
    '    at runMicrotasks (<anonymous>)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n' +
    '    at async Promise.all (index 1)',
  [message]: 'Invalid key length',
  code: 'ERR_CRYPTO_INVALID_KEYLEN'
} 
```

My couchdb secret is `6c1953b6-e64d-4b0c-9268-2528396f2f58`. 

The internet says that the buffer length is the problem. I hacked by by doing: 
```
const cipher = crypto.createCipheriv(CRYPTO_ALGO, Buffer.from(key).slice(0, 32), iv);
const decipher = crypto.createDecipheriv(CRYPTO_ALGO, Buffer.from(key).slice(0, 32), iv);
```

Reply:
@dianabarsan Great catch, thanks! Fixed and integration test added.